src/Symfony/Component/Form/Extension/Core/EventListener/ResizeFormListener.php

```diff
@@ -102,11 +102,15 @@ public function preSubmit(FormEvent $event)
         $form = $event->getForm();
         $data = $event->getData();
 
+        if ($data instanceof \Traversable){
+            @trigger_error('Support for Traversable is deprecated since version 3.1 and will be removed in 4.0. You should not use it anymore.', E_USER_DEPRECATED);
+        }
+
         if (null === $data || '' === $data) {
             $data = array();
         }
 
-        if (!is_array($data) && !($data instanceof \Traversable && $data instanceof \ArrayAccess)) {
```

Comment:
Actually, you cannot remove this check because would be a BC break, you only need to add the deprecation.